### PR TITLE
refactor(allowance): enable action

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Switching allowance/enable to TS

### Link this PR to an issue

Relates to #989

### Type of change (Remove other not matching type)

- `refactor`: Migration to TypeScript

### How has this been tested?

Added tests

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
No need to do it here but do you think we should/can type `const mockAllowances = [...`

**Comment by im-adithya** [comment]
The `bulkAdd` accepts only Allowance[] type, so it gives off errors anyway. But yeah, we should type it as per practice!

**Review by escapedcat — APPROVED** [review]
Tested

File: src/extension/background-script/actions/allowances/__tests__/enable.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      args: {
+        host: "pro.kollider.xyz",
+      },
+    };
+
+    const sender: Runtime.MessageSender = {};
+
+    expect(await enableAllowance(message, sender)).toStrictEqual({
+      data: {
+        enabled: true,
```
Comment (lisabaut) [inline src/extension/background-script/actions/allowances/__tests__/enable.test.ts:119]: Why not check here for `remember` too? 🤔

**Review by lisabaut — APPROVED** [review]
Thanks! 
Just one question of understanding below.

File: src/extension/background-script/actions/allowances/__tests__/enable.test.ts (hunk 1 above)
Comment (im-adithya) [reply src/extension/background-script/actions/allowances/__tests__/enable.test.ts:119]: It's the test case for an already enabled allowance, in that case we don't open the prompt and hence don't get the `remember` info.

Comment (lisabaut) [reply src/extension/background-script/actions/allowances/__tests__/enable.test.ts:119]: 👍

**Pull request merged**